File: frontend/src/api/classroom.ts
import { http } from '@/utils/request'
import type { IPage } from './types'

// 教室接口类型定义
export interface Classroom {
  id?: number
  classroomNo?: string
  classroomName?: string
  schoolId?: number
  building?: string
  floor?: string
  capacity?: number
  classroomType?: string
  hasProjector?: number
  hasComputer?: number
  hasMultimedia?: number
  equipment?: string
  status?: number
  createTime?: string
  updateTime?: string
  // 关联查询字段
  school?: {
    id: number
    schoolName: string
  }
  schoolName?: string
}

// 分页查询参数
export interface ClassroomPageParams {
  current: number
  size: number
  schoolId?: number
  keyword?: string
}

/**
 * 获取教室分页列表
 */
export function getClassroomPageApi(params: ClassroomPageParams) {
  return http.get<IPage<Classroom>>('/classroom/page', { params })
}

/**
 * 获取所有教室列表
 */
export function getClassroomListApi(schoolId?: number) {
  return http.get<Classroom[]>('/classroom/list', {
    params: schoolId ? { schoolId } : undefined
  })
}

/**
 * 根据ID获取教室详情
 */
export function getClassroomByIdApi(id: number) {
  return http.get<Classroom>(`/classroom/${id}`)
}

/**
 * 创建教室
 */
export function createClassroomApi(data: Classroom) {
  return http.post<void>('/classroom', data)
}

/**
 * 更新教室
 */
export function updateClassroomApi(data: Classroom) {
  return http.put<void>('/classroom', data)
}

/**
 * 删除教室
 */
export function deleteClassroomApi(id: number) {
  return http.delete<void>(`/classroom/${id}`)
}

/**
 * 获取教室使用统计
 */
export function getClassroomStatisticsApi(id: number) {
  return http.get<any>(`/classroom/${id}/usage-statistics`)
}

/**
 * 查询可用教室
 */
export function getAvailableClassroomsApi(params: {
  schoolId?: number
  minCapacity?: number
  hasProjector?: number
  hasComputer?: number
}) {
  return http.get<Classroom[]>('/classroom/available', { params })
}

File: backend/src/main/java/com/teaching/employment/entity/Classroom.java
package com.teaching.employment.entity;

import com.baomidou.mybatisplus.annotation.*;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 教室实体类
 *
 * @author Teaching Employment Platform Team
 * @since 2026-01-12
 */
@Data
@TableName("t_classroom")
@ApiModel(description = "教室信息")
public class Classroom implements Serializable {

    private static final long serialVersionUID = 1L;

    @ApiModelProperty(value = "教室ID")
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    @ApiModelProperty(value = "教室编号")
    private String classroomNo;

    @ApiModelProperty(value = "教室名称")
    private String classroomName;

    @ApiModelProperty(value = "所属学校ID")
    private Long schoolId;

    @ApiModelProperty(value = "所在楼栋")
    private String building;

    @ApiModelProperty(value = "楼层")
    private String floor;

    @ApiModelProperty(value = "容纳人数")
    private Integer capacity;

    @ApiModelProperty(value = "教室类型")
    private String classroomType;

    @ApiModelProperty(value = "是否有投影仪：1-是 0-否")
    private Integer hasProjector;

    @ApiModelProperty(value = "是否有电脑：1-是 0-否")
    private Integer hasComputer;

    @ApiModelProperty(value = "是否多媒体教室：1-是 0-否")
    private Integer hasMultimedia;

    @ApiModelProperty(value = "设备描述")
    private String equipment;

    @ApiModelProperty(value = "状态：1-可用 0-不可用")
    private Integer status;

    @ApiModelProperty(value = "是否删除：1-已删除 0-未删除")
    @TableLogic
    private Integer isDeleted;

    @ApiModelProperty(value = "创建时间")
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @ApiModelProperty(value = "更新时间")
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    /**
     * 学校信息（关联查询时使用，不映射到数据库字段）
     */
    @TableField(exist = false)
    private School school;

    /**
     * 学校名称（关联查询时使用，不映射到数据库字段）
     */
    @TableField(exist = false)
    private String schoolName;
}
